backend/models/distance.py

```diff
@@ -188,6 +188,42 @@ def measure_between_two_points(self, point_a, point_b):
             return haversine(tuple(point_a), tuple(point_b))
         return
 
+    def get_matrix(self, source_points, destination_points):
+        """
+        Retrieve a time matrix from the distance API.
+
+        Results in the matrix are returned in seconds.
+        Expects points as (latitude, longitude) tuples to be consistent with other functions.
+        NOTE - MapBox API requires a list of lon,lat;lon,lat points.
+        """
+        request = (
+            'https://api.mapbox.com/directions-matrix/v1/mapbox/driving/'
+            '{points}?sources={sources}&destinations={destinations}&access_token={token}'
+        )
+        all_points = source_points + destination_points
+        points = ';'.join(
+            '{longitude},{latitude}'.format(longitude=point[1], latitude=point[0])
+            for point in all_points
+        )
+        sources = ';'.join(
```

Comment:
Note that AFAIK you can submit these as parameters instead of manually converting to strings:

`params = {'sources': [idx for idx, _ in enumerate(source_points)], 'destinations' : [...])`